bindings/python/src/pre_tokenizers.rs

```diff
@@ -489,17 +489,67 @@ impl PyMetaspace {
         setter!(self_, Metaspace, add_prefix_space, add_prefix_space);
     }
 
+    #[getter]
+    fn get_prepend_scheme(self_: PyRef<Self>) -> String {
+        // Assuming Metaspace has a method to get the prepend_scheme as a string
+        let scheme: PrependScheme = getter!(self_, Metaspace, get_prepend_scheme());
+        match scheme {
+            PrependScheme::First => "First",
+            PrependScheme::Never => "Never",
+            PrependScheme::Always => "Always",
+        }
+        .to_string()
+    }
+    const UNKNOWN_VARIANT_ERROR_MESSAGE: &str =
+        "is an unknown variant, should be one of ['First', 'Never', 'Always']";
+
+    #[setter]
+    fn set_prepend_scheme(self_: PyRef<Self>, prepend_scheme: &str) -> PyResult<()> {
+        // Assuming Metaspace has a method to set the prepend_scheme from a string
+        let scheme = match prepend_scheme {
+            "First" => PrependScheme::First,
+            "Never" => PrependScheme::Never,
+            "Always" => PrependScheme::Always,
+            _ => {
+                return Err(exceptions::PyValueError::new_err(format!(
+                    "{} {}",
+                    prepend_scheme,
+                    Self::UNKNOWN_VARIANT_ERROR_MESSAGE,
+                )));
+            }
+        };
+        setter!(self_, Metaspace, @set_prepend_scheme, scheme);
+        Ok(())
+    }
+
     #[new]
-    #[pyo3(signature = (replacement = PyChar('▁'), add_prefix_space = true, **_kwargs), text_signature = "(self, replacement=\"_\", add_prefix_space=True)")]
+    #[pyo3(signature = (replacement = PyChar('▁'), add_prefix_space = true, prepend_scheme=None, **_kwargs), text_signature = "(self, replacement=\"_\", add_prefix_space=True)")]
     fn new(
         replacement: PyChar,
         add_prefix_space: bool,
+        prepend_scheme: Option<String>,
         _kwargs: Option<&PyDict>,
-    ) -> (Self, PyPreTokenizer) {
-        (
-            PyMetaspace {},
-            Metaspace::new(replacement.0, add_prefix_space).into(),
-        )
+    ) -> PyResult<(Self, PyPreTokenizer)> {
+        // Create a new Metaspace instance
+        let mut new_instance: Metaspace = Metaspace::new(replacement.0, add_prefix_space);
+
+        // If a prepend scheme is provided, set it
+        if let Some(prepend_scheme) = prepend_scheme {
+            let prepend_scheme_enum = match prepend_scheme.as_str() {
+                "First" => PrependScheme::First,
```

Comment:
Please factor this code out ! :D. You shouldn't need to have this twice, Make it a simple function.

Also why change `"First"`  ? `"first"` is much more pythonic imo.